/*

***********************************************************************************
* Copyright (C) 2018 - 2020, BlockSettle AB
* Distributed under the GNU Affero General Public License (AGPL v3)
* See LICENSE or http://www.gnu.org/licenses/agpl.html
*
**********************************************************************************

*/
syntax = "proto3";

package bs.types;

enum AssetType
{
   ASSET_TYPE_UNKNOWN = 0;
   ASSET_TYPE_SPOT_FX = 1;
   ASSET_TYPE_SPOT_XBT = 2;
   ASSET_TYPE_PRIVATE_MARKET = 3;
}

enum Side
{
    SIDE_INVALID = 0;
    SIDE_BUY = 1;
    SIDE_SELL = 2;
}

enum OrderStatus
{
   ORDER_STATUS_UNKNOWN = 0;
   ORDER_STATUS_PENDING = 1;
   ORDER_STATUS_FILLED = 2;
   ORDER_STATUS_VOID = 3;
}

message Order
{
   OrderStatus status = 2;
   string status_text = 3;
   int64 timestamp_ms = 4;
   string product = 5; // for example "XBT"
   string product_against = 6; // for example "EUR"
   Side side = 7;
   double quantity = 8;
   double price = 9;
}

message ChatToken
{
   int64 timestamp_ms = 1;

   string chat_login = 2;

   // bs::network::UserType as int
   int32 user_type = 3;
}

message Balance
{
    string currency = 1;
    double balance = 2;
}

message TradeSettings
{
    uint64 xbt_tier1_limit = 1;
    uint32 xbt_price_band = 2;
    uint32 auth_required_settled_trades = 3;
}
